php/libraries/Database.class.inc

```diff
@@ -988,6 +987,34 @@ class Database
         return $result;
     }
 
+    /**
+     * Helper function used to select a single cell from the database and convert
+     * it to an integer.
+     *
+     * @param string   $query  The SQL statement to run
+     * @param string[] $params Values to use for binding in the prepared statement
+     *
+     * @return ?int A cell from the database converted to an integer value. If
+     * the query returns a value that cannot be validated as an integer, the
+     * function returns null. The calling code is responsible for handling the
+     * null case appropriately.
+     *
+     * @throws \DomainException When the cell returned by the query cannot be
+     *                          validated as an integer.
+     */
+    function pselectOneInt(string $query, array $params): ?int
+    {
+        $result = filter_var(
```

Comment:
apparently `FILTER_VALIDATE_INT` does not interpret 0 as an int so a special case needs to be added for the 0 case

Reply:
> As of PHP 5.4.11, the numbers +0 and -0 validate as both integers as well as floats (using FILTER_VALIDATE_FLOAT and FILTER_VALIDATE_INT). Before PHP 5.4.11 they only validated as floats (using FILTER_VALIDATE_FLOAT). 

From PHP manual https://www.php.net/manual/en/filter.filters.validate.php

Where did you get this info from? did it fail manual tests?

Reply:
https://www.w3schools.com/php/filter_validate_int.asp

Reply:
I ran their example code and it seems to work fine. What happens when you test it? it's possible that article was written before 5.4.11 -- I can't find a date on the page